.github/workflows/book.yml

```diff
@@ -0,0 +1,61 @@
+on:
+  push:
+     branches:
+       - master
+       - develop
+       - release/* 
+     tags:
+       - v1
+       - v1*
+
+# render book
+name: renderbook
+
+jobs:
+  bookdown:
+    name: Render-Book
+    runs-on: ubuntu-latest
+    container: pecan/base:latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: r-lib/actions/setup-r@v1
+      - uses: r-lib/actions/setup-pandoc@v1
+      - name: Install rmarkdown
+        run: Rscript -e 'install.packages(c("rmarkdown","bookdown"))'
+      - name: Render Book
+        run: cd book_source && Rscript -e 'options(bookdown.render.file_scope=FALSE); bookdown::render_book("index.Rmd")'
+      - uses: actions/upload-artifact@v2
```

Comment:
@MukulMaheshwari it's fine to use artifacts, but it will be better if we use [actions/cache](https://github.com/marketplace/actions/cache).